docs/constrained-generators/manual.md

```diff
@@ -1,28 +1,109 @@
+    - [From Term to Pred](#from-term-to-pred)
+    - [For all elements in a container type (List, Set, Map)](#for-all-elements-in-a-container-type-list-set-map)
+    - [Reification](#reification)
+    - [Disjunction, choosing between multiple things with the same type](#disjunction-choosing-between-multiple-things-with-the-same-type)
+    - [Primed library functions which are compositions with match](#primed-library-functions-which-are-compositions-with-match)
+    - [Constructors and Selectors](#constructors-and-selectors)
+    - [Naming introduced lambda bound Term variables](#naming-introduced-lambda-bound-term-variables)
+    - [Existential quantifiers](#existential-quantifiers)
+    - [Conditionals](#conditionals)
+    - [Explanations](#explanations)
+    - [Operations to define and use Specifications](#operations-to-define-and-use-specifications)
+    - [Utility functions](#utility-functions)
+    - [Escape Hatch to QuickCheck Gen monad](#escape-hatch-to-quickcheck-gen-monad)
+- [Strategy for constraining a large type with many nested sub-components.](#strategy-for-constraining-a-large-type-with-many-nested-sub-components)
+- [Writing HasSpec instances by hand.](#writing-hasspec-instances-by-hand)
+    - [Strategy 1 using GHC.Generics](#strategy-1-using-ghcgenerics)
+    - [Strategy 2 writing your own SimpleRep instance](#strategy-2-writing-your-own-simplerep-instance)
+    - [Strategy 3 defining the SimpleRep instance in terms of another type with a SimpleRep instance](#strategy-3-defining-the-simplerep-instance-in-terms-of-another-type-with-a-simplerep-instance)
+    - [Strategy 4, bypassing SimpleRep, and write the HasSpec instance by Hand](#strategy-4-bypassing-simplerep-and-write-the-hasspec-instance-by-hand)
+- [A look into the internals of the system.](#a-look-into-the-internals-of-the-system)
+
+<!-- markdown-toc end -->
+
+
 ## Constrained Generators is a First-Order Logic
 
 
 A First-order typed logic has 4 components
 
 1. **Terms** consisting of 
-  - Variables (i.e. x), 
-  - Constants (i.e. 5), 
-  - Applications (ie. F x 5). Applications, apply a function symbol to a list of arguments: (i.e.  FunctionSymbol term1 .. termN)
-2. **Predicates**   (x == 5)
-3. **Connectives**  (And, Or, Not, =>, ...)
+   - Variables: `x`, `y` . 
+   - Constants: `5`, `"abc"`, `True` . 
+   - Applications: `elem_ "abc" xs`.  Applications apply a function symbol (i.e. `elem_`) to a list of Terms
+2. **Predicates**   (Assert (x ==. 5)). Predicates are the assertions of boolean typed terms.
+3. **Connectives**  (And, Or, Not, =>, ...).  Connectives make complicated Predicates out of simpler ones.
 4. **Quantifiers**  (Forall, Exists)
 
 The **Constrained generators** system allows programmers to write Haskell programs with type `(Specification T)` that denotes a set of random values for the type `T`, that are subject to a set of constraints. This supports property based testing where a completely random set of values may not be useful.
```

Comment:
Maybe remove the parenthesis around the type (since it is the object of the phrase).
```suggestion
The **Constrained generators** system allows programmers to write Haskell programs with type `Specification T` that denotes a set of random values for the type `T`, that are subject to a set of constraints. This supports property based testing where a completely random set of values may not be useful.
```

Reply:
done